test/core/tsi/crl_ssl_transport_security_test.cc

```diff
@@ -185,12 +238,27 @@ class CrlSslTransportSecurityTest
       // For OpenSSL versions < 1.1, TLS 1.3 is not supported, so the
       // client-side handshake should succeed precisely when the server-side
       // handshake succeeds.
+      //
+      // For the intermediate cases, we have a CA -> Intermediate CA -> Leaf
+      // Cert chain in which the Intermediate CA cert is revoked by the CA. We
+      // test 3 cases. Note: A CRL not existing should not make the handshake
+      // fail
+      // 1. CRL Directory with CA's CRL and Intermediate CA's CRL -> Handshake
+      // fails due to revoked cert
+      // 2. CRL Directory with CA's CRL but missing Intermediate CA's CRL ->
+      // Handshake fails due to revoked cert
+      // 3. CRL Directory without CA's CRL with but Intermediate CA's CRL ->
+      // Handshake succeeds because the CRL that revokes the cert is not
+      // present.
       bool expect_server_success =
-          !(use_revoked_server_cert_ || use_revoked_client_cert_);
+          !(use_revoked_server_cert_ || use_revoked_client_cert_ ||
```

Comment:
I think it'd be cleaner with
`!use_revoked_server_cert_ && !use_revoked_client_cert && !(use_revoked_intermediate_ && !use_missing_root_crl_);`. WDYT? Same below.

nit: I guess `&` for bit operation works well here as well but normally it should be `&&` for logical `and`.

Reply:
I think it's just a personal preference - the way I read it I see something like
"expect server success = NOT any of the revoked bools being true" So if anything bool flag is true in all of those ORs it'll be an expected failure.

IMO I like the way it is a little more because I don't have the think about and negate every individual condition

Overall, I'd rather pull some of this shared logic out and hard code it in each individual test since it's the known expected result, I think it would make it a lot cleaner, but that's a refactor for later.

Reply:
Sure. I don't have a strong opinion either.